contracts/src/v0.8/functions/dev/1_0_0/FunctionsCoordinator.sol

```diff
@@ -0,0 +1,250 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.6;
+
+import {IFunctionsCoordinator} from "./interfaces/IFunctionsCoordinator.sol";
+import {IFunctionsBilling, FunctionsBilling, Route, ITypeAndVersion} from "./FunctionsBilling.sol";
+import {OCR2Base} from "./ocr/OCR2Base.sol";
+import {Functions} from "./Functions.sol";
+import {IOwnable} from "../../../shared/interfaces/IOwnable.sol";
+import {IFunctionsRouter} from "./interfaces/IFunctionsRouter.sol";
+import {IFunctionsSubscriptions} from "./interfaces/IFunctionsSubscriptions.sol";
+
+/**
+ * @title Functions Coordinator contract
+ * @notice Contract that nodes of a Decentralized Oracle Network (DON) interact with
+ * @dev THIS CONTRACT HAS NOT GONE THROUGH ANY SECURITY REVIEW. DO NOT USE IN PROD.
+ */
+contract FunctionsCoordinator is OCR2Base, IFunctionsCoordinator, FunctionsBilling {
+  uint16 constant REQUEST_DATA_VERSION = Functions.REQUEST_DATA_VERSION;
+
+  event OracleRequest(
+    bytes32 indexed requestId,
+    address requestingContract,
+    address requestInitiator,
+    uint64 subscriptionId,
+    address subscriptionOwner,
+    bytes data
+  );
+  event OracleResponse(bytes32 indexed requestId);
+  event UserCallbackError(bytes32 indexed requestId, string reason);
+  // event RawError(bytes32 indexed requestId, bytes lowLevelData); TODO
+  event InvalidRequestID(bytes32 indexed requestId);
+  event ResponseTransmitted(bytes32 indexed requestId, address transmitter);
+  event InsufficientGasProvided(bytes32 indexed requestId);
```

Comment:
Also these 3 events, not sure we really need them for the node (the listener).

Reply:
Not for the node but do we still want them for observability?

Reply:
Some of these we may want to alert on, for example if `InsufficientGasProvided` ever happens that means either something went wrong in the core node, or the transmitter underfunded the gas on purpose

Reply:
We never used `InvalidRequestID`, I suggest to remove it.
`ResponseTransmitted` is now combined with RequestEnd and therefore needs to be removed.
`InsufficientGasProvided` - we can keep this one if you believe we will need an alert.

Reply:
When `InsufficientGasProvided` is emitted, do we emit `RequestEnd`?